### ISSUE TITLE
Code Table Request - identifiers - DOI

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#6442]
## Initial Request

DOI [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=CTCOLL_OTHER_ID_TYPE#doi) including base URL ---> identifier

Data:

[temp_doi.csv.zip](https://github.com/ArctosDB/arctos/files/11837549/temp_doi.csv.zip)

Summary:

```

 guid_prefix | numrecs 
-------------+---------
 DMNS:Mamm   |     258
 MSB:Para    |       1
 UCM:Herp    |      66

```

 @ebraker
 @msbparasites
 @campmlc
 @acdoll
 @jrdemboski

@ArctosDB/arctos-code-table-administrators 

## Approval

_All of the following must be checked before this may proceed._

_The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality). **No person should act in multiple roles**; the submitter cannot also serve as a Code Table Administrator, for example._

- [x] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [x] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection

_If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc._

1. _Can a suitable solution be found here? If not, proceed to (2)_
2. _Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)_
3. _Take the discussion to a monthly Arctos Working Group meeting for final resolution._

## Implementation

_Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made._

_Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel._

_Make changes as described above. Ensure the URL of this Issue is included in the definition._

_Close this Issue._

_**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result._

## Special Exemptions

_In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases._

1. _Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step._
2. _The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology._

**Comment by acdoll** [comment]
So... it seems that these records just need 'https://datadryad.org/stash/dataset/doi:' affixed to the beginning of these values, correct?

**Comment by mkoo** [comment]
Well the Issued by should be Dryad, right? DOI's can be minted by anyone so we need to know who that agent is.
do you have an example record I could see pleaes? @acdoll

**Comment by dustymc** [comment]
>So... it seems that these records just need 'https://datadryad.org/stash/dataset/doi:' affixed to the beginning of these values, correct?

No.

If that happens, then https://datadryad.org/stash/dataset/doi:10.5061/dryad.b554m44 dies the second Dryad fails to pay their bill and their URL gets bought by some tree nymph aficionado.

These need prepended with https://doi.org/ (==> https://doi.org/10.5061/dryad.b554m44 ) - that's identical now, but the DOI folks can (and largely exists to) just redirect to dryad's new URL once ^^ that happens. 

That said, Dryad has done an exceptional job of making their URL transparent and I could definitely figure out what to do with https://datadryad.org/stash/dataset/doi:10.5061/dryad.b554m44 in the event of some catastrophe - the details probably aren't terribly important in this particular case.

That should probably be documented (in collection's procedures) so new records are also entered in the most stable way possible.

**Comment by acdoll** [comment]
> an example record

https://arctos.database.museum/guid/DMNS:Mamm:12664




> These need prepended with https://doi.org/

So is this done already? They all seem to show the link appropriately:![image](https://github.com/ArctosDB/arctos/assets/12507297/bfb23091-16a6-4355-b6d0-f654a520feda)

**Comment by dustymc** [comment]
>So is this done already? They all seem to show the link appropriately:

No, but that's the general trend here: Same data, same functionality, just simpler to use and fewer weeds to get lost in - and the potential for carrying a lot more data, but I can't magick it out of the æther so that's not important yet. 

(And I think maybe that's what @mkoo has been trying to tell me - these should all be treated as internal cleanup, because they are, I should just do these, the Issues are just a distraction....)

**Comment by mkoo** [comment]
@dustymc ahem (yes)
@acdoll that link works! But issuer should be Dryad, yes?
@dustymc can you find all the idtypes = doi where the id value contains "dryad" ? or is DMNS the only one recording these at the moment. Either way-- best to fix for all. Nothing lost, all gain in cleaning this up!

**Comment by dustymc** [comment]
I'm a little slow on the uptake maybe - but check a box up yonder and we'll be done here...

Data above, there are some UCM, they're all dryad - and I can add that, but issuer is less-important (for my stuff, maybe not some coup-counting in the collections) for resolvable IDs so I haven't been worried too much there.

**Comment by campmlc** [comment]
I'm OK with adding Dryad as the issuer. But the proposal is to get rid of DOI as the identifier type and replace it with identifier? So it won't be possible to search for all records that have DOIs.

**Comment by dustymc** [comment]
>So it won't be possible to search for all records that have DOIs.

That is incorrect.

**Comment by campmlc** [comment]
Please explain how we would search for all records that have dois if DOI is no longer an identifer type? I am honestly asking for clarification.

**Comment by Jegelewicz** [comment]
I'm just guessing, but search identifier value = "https://doi.org/%"

**Comment by mkoo** [comment]
I think it actually makes sense to keep idtype of ```doi``` because as I said a lot of entities can mint doi's not just Dryad. (I mean Arctos does too!)

**Comment by dustymc** [comment]
> lot of entities can mint doi's

That (and the data in https://github.com/ArctosDB/arctos/issues/6443) is a compelling reason to **not** try to type this! DOI-ness isn't important - anyone can get one and use it for anything. Arctos and Dryad both issue them. (And knowing that doesn't provide any information in either case!) Dryads are "data" which could be ANYTHING, Arctos' are something in some random node.

Zoom?

**Comment by mkoo** [comment]
(who are you trying to zoom?! me? probably...)

OK, I see your point *if squinting* but I also know that there is value to have it explicit. Smithsonian has catalog numbers, doi, probably ark's, etc so make it explicit what **type** they are. It is called IDtype after all. 

Convince me better! (and yes send me an email about tomorrow)

**Comment by Jegelewicz** [comment]
See also https://github.com/ArctosDB/arctos/issues/6443#issuecomment-1604339133

**Comment by mkoo** [comment]
Salient facts that came out of our zoom call:
- Idtypes are **not** needed for prepends because they get in the way. So use full link for the value! We can tweak data entry UI so it's clear what will work.
- Arctos can handle the difference between values which are URL link and numbers. The URl s can be used to create pretty links
- search UI can modified so searching for DOI, ARK, etc will be easier. Mock-ups to come
